src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
+
+#include <algorithm>
+
+//--------------------------------------------------------------------------------------------------
+// Container-wide versions of std algoithms.
+//--------------------------------------------------------------------------------------------------
+
+template <typename Container, typename Predicate>
+inline bool any_of(Container&& c, Predicate p) {
+    using std::begin; //ADL
```

Comment:
These are supremely uninformative comments :/
One, either you know what it's referencing and you don't need the comment, or you have to go look up what it means.
Two, merely noting that it has something to do with ADL doesn't tell you that, "we're adding std::begin to the local namespace so that if there's a local begin that should take precedence it is used instead of std::begin."